src/lib/npm-helper.js

```diff
@@ -0,0 +1,202 @@
+    }
+
+    if (!tagOrVersion) {
+      tagOrVersion = 'latest'
+    }
+  } else if (spec.startsWith('file:') || spec.includes('/') || spec.includes('\\')) { // process file paths
+    const fileProtocol = 'file:'
+    let filePath = spec.startsWith(fileProtocol) ? spec.substring(fileProtocol.length) : spec
+
+    // all absolute paths need to be converted to relative paths
+    if (path.isAbsolute(filePath)) {
+      filePath = path.relative(dir, filePath)
+    }
+
+    url = `file:${filePath}`
+  } else { // it's a plain package name
+    name = spec
+    tagOrVersion = 'latest'
+  }
+
+  return { url, name, tagOrVersion }
+}
+
+/** @private */
+async function readPackageJson (dir = process.cwd()) {
+  const filePath = path.join(dir, 'package.json')
+  return fs.readJson(filePath)
+}
+
+/** @private */
+async function writeObjectToPackageJson (obj, dir = process.cwd()) {
+  const filePath = path.join(dir, 'package.json')
+  const pkgJson = await fs.readJson(filePath)
+
+  return fs.writeJson(
+    filePath,
+    { ...pkgJson, ...obj },
+    { spaces: 2 }
+  )
+}
+
+/** @private */
+async function getNpmDependency ({ packageName, urlSpec }, dir = process.cwd()) {
+  // go through package.json and find the key for the urlSpec
+  const packageJson = await readPackageJson(dir)
+  aioLogger.debug(`getNpmPackageName package.json: ${JSON.stringify(packageJson, null, 2)}`)
+
+  if (packageName) {
+    return Object.entries(packageJson.dependencies || {})
+      .find(([key, value]) => {
+        aioLogger.debug(`k,v: ${key}, ${value}`)
+        return key === packageName
+      })
+  } else if (urlSpec) {
+    return Object.entries(packageJson.dependencies || {})
+      .find(([key, value]) => {
+        aioLogger.debug(`k,v: ${key}, ${value}`)
+        return value === urlSpec
+      })
+  }
+
+  throw new Error('Either packageName or urlSpec must be set')
+}
+
+/**
+ * Gets the latest version of a plugin from npm.
+ *
+ * @param {string} npmPackageName the npm package name of the plugin
+ * @returns {string} the latest version of the plugin from the npm registry
+ */
+async function getNpmLatestVersion (npmPackageName) {
+  const res = await fetch(`https://registry.npmjs.com/${npmPackageName}`)
+  const { 'dist-tags': distTags } = await res.json()
+  return distTags && distTags.latest
+}
+
+/**
+ * Gets the npm package version of an npm package installed in the cli.
+ *
+ * @param {string} npmPackageName the npm package name
+ * @param {string} dir the root path of where node_modules is
+ * @returns {string} the version of the package from the cli node_modules
+ */
+async function getNpmLocalVersion (npmPackageName, dir = process.cwd()) {
+  const pjsonPath = path.join(dir, 'node_modules', npmPackageName, 'package.json')
+  const pjson = JSON.parse(fs.readFileSync(pjsonPath))
+
+  return pjson.version
+}
+
+/**
+ * Hide NPM Warnings by intercepting process.stderr.write stream
+ *
+ */
+function hideNPMWarnings () {
```

Comment:
This seems like a bad practice, and has side-effects.
Can't we just use --loglevel=error for the `npm install`

Reply:
this was a copy paste from the discover plugin - to suppress warnings one level deeper that we have no control over (the plugin install command). Yes we should definitely change to what you recommended